From 22a821947f6a6621b8a8bb73e42b0b2cf9c3a6ca Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Thu, 20 Oct 2011 11:29:39 +0800
Subject: [PATCH] rk29: ddr3sdk: do not define empty
 rk29_setgpio_suspend_board/rk29_setgpio_resume_board, which is weak function

---
 arch/arm/mach-rk29/board-rk29-ddr3sdk.c | 59 +++++++++++--------------
 1 file changed, 26 insertions(+), 33 deletions(-)

diff --git a/arch/arm/mach-rk29/board-rk29-ddr3sdk.c b/arch/arm/mach-rk29/board-rk29-ddr3sdk.c
index ff7a27e43658..8499e1e44caa 100755
--- a/arch/arm/mach-rk29/board-rk29-ddr3sdk.c
+++ b/arch/arm/mach-rk29/board-rk29-ddr3sdk.c
@@ -381,44 +381,37 @@ struct platform_device rk29_device_dma_cpy = {
 #endif
 
 #if defined(CONFIG_RK29_GPIO_SUSPEND)
-void key_gpio_pullupdown_enable(void)
-{	 gpio_pull_updown(RK29_PIN6_PA0, 0);
-	 gpio_pull_updown(RK29_PIN6_PA1, 0);
-	 gpio_pull_updown(RK29_PIN6_PA2, 0);
-	 gpio_pull_updown(RK29_PIN6_PA3, 0);
-	 gpio_pull_updown(RK29_PIN6_PA4, 0);
-	 gpio_pull_updown(RK29_PIN6_PA5, 0);
-	 gpio_pull_updown(RK29_PIN6_PA6, 0);
-
-}
-void key_gpio_pullupdown_disable(void)
-{
-	 gpio_pull_updown(RK29_PIN6_PA0, 1);
-	 gpio_pull_updown(RK29_PIN6_PA1, 1);
-	 gpio_pull_updown(RK29_PIN6_PA2, 1);
-	 gpio_pull_updown(RK29_PIN6_PA3, 1);
-	 gpio_pull_updown(RK29_PIN6_PA4, 1);
-	 gpio_pull_updown(RK29_PIN6_PA5, 1);
-	 gpio_pull_updown(RK29_PIN6_PA6, 1);
-}
-void  rk29_setgpio_suspend_board(void)
-{	
-	
-	key_gpio_pullupdown_enable();
-	
+static void key_gpio_pullupdown_enable(void)
+{
+	gpio_pull_updown(RK29_PIN6_PA0, 0);
+	gpio_pull_updown(RK29_PIN6_PA1, 0);
+	gpio_pull_updown(RK29_PIN6_PA2, 0);
+	gpio_pull_updown(RK29_PIN6_PA3, 0);
+	gpio_pull_updown(RK29_PIN6_PA4, 0);
+	gpio_pull_updown(RK29_PIN6_PA5, 0);
+	gpio_pull_updown(RK29_PIN6_PA6, 0);
 }
 
-void  rk29_setgpio_resume_board(void)
-{	 
-	
-	 key_gpio_pullupdown_disable();  
+static void key_gpio_pullupdown_disable(void)
+{
+	gpio_pull_updown(RK29_PIN6_PA0, 1);
+	gpio_pull_updown(RK29_PIN6_PA1, 1);
+	gpio_pull_updown(RK29_PIN6_PA2, 1);
+	gpio_pull_updown(RK29_PIN6_PA3, 1);
+	gpio_pull_updown(RK29_PIN6_PA4, 1);
+	gpio_pull_updown(RK29_PIN6_PA5, 1);
+	gpio_pull_updown(RK29_PIN6_PA6, 1);
+}
 
- }
-#else
 void rk29_setgpio_suspend_board(void)
-{}
+{
+	key_gpio_pullupdown_enable();
+}
+
 void rk29_setgpio_resume_board(void)
-{}
+{
+	key_gpio_pullupdown_disable();
+}
 #endif
 
 #if defined(CONFIG_RK_IRDA) || defined(CONFIG_BU92747GUW_CIR)
-- 
2.35.3

